spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -183,4 +187,22 @@ private AppDefinition mergeAndExpandAppProperties(TaskDefinition original, Resou
 		return new AppDefinition(original.getName(), merged);
 	}
 
+	private TaskDefinition updateTaskProperties(TaskDefinition taskDefinition, TaskExecution taskExecution) {
+		TaskDefinitionBuilder builder = TaskDefinitionBuilder.from(taskDefinition);
+		builder.setProperty("spring.datasource.url", dataSourceProperties.getUrl());
+		builder.setProperty("spring.datasource.username",
+			dataSourceProperties.getUsername());
+		// password may be empty
+		if (StringUtils.hasText(dataSourceProperties.getPassword())) {
+			builder.setProperty("spring.datasource.password",
+				dataSourceProperties.getPassword());
+		}
+		builder.setProperty("spring.datasource.driverClassName",
+			dataSourceProperties.getDriverClassName());
+
+		builder.setProperty("spring.cloud.task.executionid", String.valueOf(taskExecution.getExecutionId()));
```

Comment:
One simple case is if the user wanted to use pass in the commandLine arg `executionId` for their own application. The prefix of spring.cloud.task marks the executionID specifically for the use by Spring Cloud Task.